plugins/inventory/vmware_vm_inventory.py

```diff
@@ -68,19 +69,69 @@
+            - In addition to VM properties, the following are special values
+            - Use 'customValue' to populate VM's custom attributes.
+            - Use 'all' to populate all VM properties.
             type: list
             default: [ 'name', 'config.cpuHotAddEnabled', 'config.cpuHotRemoveEnabled',
                        'config.instanceUuid', 'config.hardware.numCPU', 'config.template',
-                       'config.name', 'guest.hostName', 'guest.ipAddress',
+                       'config.name', 'config.uuid' ,'guest.hostName', 'guest.ipAddress',
                        'guest.guestId', 'guest.guestState', 'runtime.maxMemoryUsage',
-                       'customValue'
+                       'customValue', 'summary.runtime.powerState', 'config.guestId'
                        ]
+            version_added: "2.9"
+        resources:
+            description:
+            - A list of resources to limit search scope.
+            - Each resource item is represented by exactly one C('vim_type_snake_case):C(list of resource names) pair and optional nested I(resources)
+            - Key name is based on snake case of a vim type name; e.g C(host_system) correspond to C(vim.HostSystem)
+            - See  L(VIM Types,https://pubs.vmware.com/vi-sdk/visdk250/ReferenceGuide/index-mo_types.html)
+            version_added: "2.10"
+            required: False
+            type: list
+            default: []
+        with_path:
+            description:
+            - Include virtual machines path.
+            - Set this option to a string value to replace root name from I('Datacenters').
+            default: False
+            type: bool
+            version_added: "2.10"
+        hostnames:
+            description:
+                - A list of templates in order of precedence to compose inventory_hostname.
+                - Ignores template if it resulted in an empty string or None value.
+                - You can use property specified in I(properties) as variables in template.
+            type: list
+            default: ['config.name + "_" + config.uuid']
+            version_added: "2.10"
+        with_nested_properties:
+            description:
+                - This option transform flatten properties name to nested dictionary.
+            type: bool
+            default: False
+            version_added: "2.10"
+        with_sanitized_property_name:
+            description:
+                - This option allows property name sanitization to create safe property names for use in Ansible.
+                - Also, transforms property name to snake case.
+            type: bool
+            default: False
+            version_added: "2.10"
+        keyed_groups:
+            description: Add hosts to group based on the values of a variable.
+            type: list
+            default: [
+                {key: 'config.guestId', separator: ''},
+                {key: 'summary.runtime.powerState', separator: ''},
```

Comment:
this seems to rely on `summary` being included in the properties. Without it in there, I'll get a traceback that summary is not defined (when strict). That might be okay, just wanted to point it out.

Reply:
Those default values are set to maintain backward-compatible with 2.9. And yes, I have mentioned that earlier, that those values need to be reviewed. 

>- Set defaults values of options that follow ansible standard/best practices. Currently, are set for compatibility with 2.9.